tempesta_fw/cache.c

```diff
@@ -252,10 +252,17 @@ tfw_cache_employ_req(TfwHttpReq *req)
 }
 
 static bool
-tfw_cache_employ_resp(TfwHttpReq *req)
+tfw_cache_employ_resp(TfwHttpResp *resp, TfwHttpReq *req)
 {
 	if (req->cache_ctl.flags & TFW_HTTP_CC_CFG_CACHE_BYPASS)
 		return false;
+	if (!TFW_STR_EMPTY(&req->h_tbl->tbl[TFW_HTTP_HDR_AUTHORIZATION]) &&
+	    !(resp->cache_ctl.flags & TFW_HTTP_CC_PUBLIC) &&
+	    !(resp->cache_ctl.flags & TFW_HTTP_CC_S_MAXAGE) &&
+	    !(resp->cache_ctl.flags & TFW_HTTP_CC_MUST_REV))
```

Comment:
Cache control directives are out of scope of the issue and aren't correct: we should not use cached object only is age of the cache entry reaches specified max age while presence of max age doesn't mean that the response can't be cached. In particular max age equal 0 means this, so you should have been check the value.